ppsci/data/dataset/atmospheric_dataset.py

```diff
@@ -0,0 +1,1779 @@
+        num_nodes_mesh_i = mesh_i.vertices.shape[0]
+        assert np.allclose(mesh_i.vertices, mesh_ip1.vertices[:num_nodes_mesh_i])
+
+    return TriangularMesh(
+        vertices=mesh_list[-1].vertices,
+        faces=np.concatenate([mesh.faces for mesh in mesh_list], axis=0),
+    )
+
+
+def get_icosahedron():
+    phi = (1 + np.sqrt(5)) / 2
+    product = [[1.0, phi], [1.0, -phi], [-1.0, phi], [-1.0, -phi]]
+    vertices = []
+    for p in product:
+        c1 = p[0]
+        c2 = p[1]
+        vertices.append((c1, c2, 0.0))
+        vertices.append((0.0, c1, c2))
+        vertices.append((c2, 0.0, c1))
+
+    vertices = np.array(vertices, dtype=np.float32)
+    vertices /= np.linalg.norm([1.0, phi])
+
+    faces = [
+        (0, 1, 2),
+        (0, 6, 1),
+        (8, 0, 2),
+        (8, 4, 0),
+        (3, 8, 2),
+        (3, 2, 7),
+        (7, 2, 1),
+        (0, 4, 6),
+        (4, 11, 6),
+        (6, 11, 5),
+        (1, 5, 7),
+        (4, 10, 11),
+        (4, 8, 10),
+        (10, 8, 3),
+        (10, 3, 9),
+        (11, 10, 9),
+        (11, 9, 5),
+        (5, 9, 7),
+        (9, 3, 7),
+        (1, 6, 5),
+    ]
+
+    angle_between_faces = 2 * np.arcsin(phi / np.sqrt(3))
+    rotation_angle = (np.pi - angle_between_faces) / 2
+    rotation = scipy.spatial.transform.Rotation.from_euler(
+        seq="y", angles=rotation_angle
+    )
+    rotation_matrix = rotation.as_matrix()
+    vertices = np.dot(vertices, rotation_matrix)
+
+    return TriangularMesh(
+        vertices=vertices.astype(np.float32), faces=np.array(faces, dtype=np.int32)
+    )
+
+
+def get_hierarchy_of_triangular_meshes_for_sphere(
+    splits: int,
+) -> typing.List[TriangularMesh]:
+    current_mesh = get_icosahedron()
+    output_meshes = [current_mesh]
+    for _ in range(splits):
+        current_mesh = _two_split_unit_sphere_triangle_faces(current_mesh)
+        output_meshes.append(current_mesh)
+    return output_meshes
+
+
+def _two_split_unit_sphere_triangle_faces(
+    triangular_mesh: TriangularMesh,
+) -> TriangularMesh:
+    """Splits each triangular face into 4 triangles keeping the orientation."""
+    new_vertices_builder = _ChildVerticesBuilder(triangular_mesh.vertices)
+
+    new_faces = []
+    for ind1, ind2, ind3 in triangular_mesh.faces:
+        ind12 = new_vertices_builder.get_new_child_vertex_index((ind1, ind2))
+        ind23 = new_vertices_builder.get_new_child_vertex_index((ind2, ind3))
+        ind31 = new_vertices_builder.get_new_child_vertex_index((ind3, ind1))
+        new_faces.extend(
+            [
+                [ind1, ind12, ind31],  # 1
+                [ind12, ind2, ind23],  # 2
+                [ind31, ind23, ind3],  # 3
+                [ind12, ind23, ind31],  # 4
+            ]
+        )
+    return TriangularMesh(
+        vertices=new_vertices_builder.get_all_vertices(),
+        faces=np.array(new_faces, dtype=np.int32),
+    )
+
+
+class _ChildVerticesBuilder(object):
```

Comment:
python3不需要显式继承object
```suggestion
class _ChildVerticesBuilder:
```

Reply:
Done.